### PR TITLE
Travel Estimates Logic 

### PR DESCRIPTION
**Pull request by ZoroarkZark**
* Creates a functional basis for using React Query to fetch the information regarding one location's ETA from the source location (typically the user).
* Due to varying implementations, A fetch function hasn't been created at the moment to fetch the information from somewhere and return the ETA from a location. Still, support for it is available as a parameter for the hook.
* Right now, I'm open to any considerations regarding what next to do with this feature.

### CODE REVIEW

File: event-details/travel-estimates/DistanceEvaluator.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  })
+
+  const loadETAFromLocations = jest.fn()
+
+  const renderUseCalculateETA = (eventLocation: LocationCoordinate2D) => {
+    return renderHook(
+      () => useCalculateETA(queryClient, eventLocation, loadETAFromLocations),
+      {
+        wrapper: ({ children }) => (
+          <TestQueryClientProvider>{children}</TestQueryClientProvider>
```
Comment (ZoroarkZark) [inline event-details/travel-estimates/DistanceEvaluator.test.tsx:27]: Provide testQueryClient to provider, not to hook

File: event-details/travel-estimates/DistanceEvaluator.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  } else {
+    return { status: "error", data: undefined }
+  }
+}
+
+const cacheETAQueryData = (
+  etaInfo: TravelEstimates,
+  queryClient: QueryClient
+) => {
+  queryClient.setQueryData(["etaUserLocation", etaInfo.sourceLocation], etaInfo)
```
Comment (ZoroarkZark) [inline event-details/travel-estimates/DistanceEvaluator.ts:64]: Query key needs to match useUserCoordinatesQuery's query key

File: event-details/travel-estimates/DistanceEvaluator.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ */
+export const useCalculateETA = (
+  queryClient: QueryClient,
+  eventLocation: LocationCoordinate2D,
+  loadETAFromLocations: (
+    eventLocation: LocationCoordinate2D
+  ) => Promise<TravelEstimates>,
+  options?: QueryHookOptions<TravelEstimates>
+): DistanceETAData => {
+  const queryKey = ["testing-ETA", eventLocation]
```
Comment (ZoroarkZark) [inline event-details/travel-estimates/DistanceEvaluator.ts:35]: Rename to "travel-ETAs"

File: event-details/travel-estimates/DistanceEvaluator.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+type ETAInformation = {
+  localizedWarnings: string[]
+  totalSecondsFromSource: number
+}
+
+type CamelCaseRouteKeyName<Kind extends RouteKind> =
+  Kind extends "public-transport" ? "publicTransport" : Kind
+
+export type TravelEstimates = {
+  sourceLocation: LocationCoordinate2D
```
Comment (ZoroarkZark) [inline event-details/travel-estimates/DistanceEvaluator.ts:16]: userLocation: LocationObject

File: event-details/travel-estimates/DistanceEvaluator.test.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        totalSecondsFromSource: 69
+      },
+      publicTransport: {
+        localizedWarnings: [],
+        totalSecondsFromSource: 420
+      }
+    }
+  }
+
+  describe("useCalculateETA tests", () => {
```
Comment (ZoroarkZark) [inline event-details/travel-estimates/DistanceEvaluator.test.tsx:52]: Add a case for making sure the query cache is being used properly

**Review by mhayes853 — APPROVED** [review]

**Review by tifapp — APPROVED** [review]

**Pull request merged**